### ISSUE TITLE
Automate tasks like generation of derived sources i.e. in examples and run example related tests scripts

### ISSUE THREAD

**Issue by jsinger67** [jsinger67/parol#33]
Was first mentioned in (https://github.com/jsinger67/parol/pull/31#discussion_r1032767461)

**Comment by jsinger67** [comment]
All common tasks that are used could be defined using this approach.
* Format and build
* Build and test
* Build and run examples as special test scenario
* Applying version labels
* etc.

**Comment by ryo33** [comment]
cargo make is a very useful tool, but I think there are only a few projects that need it. 
Cargo is already a powerful tool for most projects.

I used it for a personal project and had the following problems.
- `cargo install cargo-make` is slow
- things are getting more and more complex

Then, I switch to `cargo` and a few portable shell scripts for CI. If we develop on Windows, we need a bash-compatible shell or WSL.

**Comment by jsinger67** [comment]
I wasn't aware of the performance problems of cargo make. Fortunately I spent no effort for this issue yet.
I will rethink the problem and find another solution.
Some things (mostly Powershell scripts that run examples) can be solved by using the built-in test infrastructure. Others (mostly generation of parser code) can be achieved by portable scripts or some kind of build steps using build.rs.

**Comment by jsinger67** [comment]
Fixed with [82fb6f4](https://github.com/jsinger67/parol/commit/82fb6f4988338c775d0a132eb75f1555dd2fdad3)
Automation with cargo-make is an add-on. You don't need to use it in regular development.
It is rather a way to automate special tasks like building example sources to check if `parol` generates different output.

If you want to use it you need to install `cargo make`:

      ```shell
      cargo install cargo-make
      ```
Then you can invoke the example source generation with this command

      ```shell
      cargo make generate_examples
      ```

The other use case "Running the generated parsers against special input files" was integrated into a  a new [integration test](https://github.com/jsinger67/parol/blob/main/crates/parol/tests/run_examples.rs).
